tabsforeditorswidget.cpp

```diff
@@ -126,27 +134,94 @@ void TabsForEditorsWidget::handlerEditorClosed(QList<Core::IEditor*> editors)
     }
 }
 
+void TabsForEditorsWidget::handleContextMenuSelected(int choice)
+{
+    switch(choice)
+    {
+    case Constants::CONTEXT_MENU_CLOSE_TAB:
+        handleTabCloseRequested(m_currentTabIndex);
+        break;
+    case Constants::CONTEXT_MENU_CLOSE_ALL_BUT_THIS_TAB:
+        handleTabCloseAllExceptOneRequested(m_currentTabIndex);
+        break;
+    case Constants::CONTEXT_MENU_CLOSE_TABS_TO_THE_RIGHT:
+        handleTabCloseToRightRequested(m_currentTabIndex);
+        break;
+    case Constants::CONTEXT_MENU_CLOSE_ALL_TABS:
+        handleTabCloseAllRequested();
+        break;
+    }
+
+}
+
+
 void TabsForEditorsWidget::handleTabCloseRequested(int index)
 {
-    if (-1 < index) {
-        QWidget *tab = m_tabWidget->widget(index);
-        if (!tab)
-            return;
-        QList<Core::IEditor*> editorsToClose;
-        editorsToClose.clear();
-        if (m_tabsEditors.contains(tab)) {
-            Core::IEditor *editor = getEditor(tab);
-            if (!editor)
+    QList<int> singleIndexList;
+    singleIndexList.append(index);
+
+    handleTabsMultipleCloseRequested(singleIndexList);
+}
+
+void TabsForEditorsWidget::handleTabsMultipleCloseRequested(QList<int>& indices)
+{
+
+    QList<Core::IEditor*> editorsToClose;
+    editorsToClose.clear();
+
+    foreach(int index, indices)
+    {
+
+        if (-1 < index) {
+            QWidget *tab = m_tabWidget->widget(index);
+            if (!tab)
                 return;
-            editorsToClose.append(editor);
 
-            Core::EditorManager::instance()->closeEditors(editorsToClose);
-            if (m_tabsEditors.contains(tab))
-                m_tabsEditors.remove(tab);
-            if (-1 < m_tabWidget->indexOf(tab))
-                m_tabWidget->removeTab(m_tabWidget->indexOf(tab));
+            if (m_tabsEditors.contains(tab)) {
+                Core::IEditor *editor = getEditor(tab);
+                if (!editor)
+                    continue;
+                editorsToClose.append(editor);
+            }
         }
     }
+
+    Core::EditorManager::instance()->closeEditors(editorsToClose);
+}
+
+void TabsForEditorsWidget::handleTabCloseToRightRequested(int index)
+{
+    QList<int> indices;
+
+    for(int i = index + 1; i < m_tabsEditors.size(); i++)
+    {
+        indices.append(i);
+    }
+
+
+    handleTabsMultipleCloseRequested(indices);
+}
+
+void TabsForEditorsWidget::handleTabCloseAllExceptOneRequested(int index)
+{
+    QList<int> indices;
+
+    for(int i = 0; i < m_tabsEditors.size(); i++)

```

Comment:
The opening curly bracket should be on the same line as condition or loop. Please update here and everywhere else.